gdk/commands/component/transformer/PublishRecipeTransformer.py

```diff
@@ -13,6 +14,14 @@ def __init__(self, _project_config: ComponentPublishConfiguration) -> None:
 
     def transform(self):
         recipe_path = Path(self.project_config.gg_build_recipes_dir).joinpath(self.project_config.recipe_file.name)
+
+        logging.info(f"Validating the file size of the built recipe {recipe_path}")
+        # Validate the size of the recipe file before processing its content.
+        valid_file_size, input_recipe_file_size = utils.is_recipe_size_valid(recipe_path)
+        if not valid_file_size:
+            logging.error(BUILT_RECIPE_SIZE_INVALID.format(self.project_config.recipe_file, input_recipe_file_size))
+            raise Exception(BUILT_RECIPE_SIZE_INVALID.format(self.project_config.recipe_file, input_recipe_file_size))
+
```

Comment:
https://github.com/aws-greengrass/aws-greengrass-gdk-cli/blob/242cf804f24b6f4292c84c07bcaa99ca2154fdb6/gdk/commands/component/transformer/PublishRecipeTransformer.py#L94 we want to add this here, after updating and before creating the recipe file.